### DISCUSSION TITLE
Feature wish: compatibility with Debian bookworm

### DISCUSSION THREAD

**Discussion by petermkr** [CLIMADA-project/climada_python#751] [General]
Hello everyone,

I have just seen that many of the version thresholds in `requirements/env_climada.yml` were resolved for good. 
Thanks for your great work!

For the team working on a containerized version of CLIMADA, this is great news as we will be able to get rid of manual compilation of libraries (e.g. proj) and reduce our build times considerably.

The official Python docker image (https://hub.docker.com/_/python/) is based on the currently stable Debian release. A new release 'bookworm' was released just some weeks ago, remaining in place for two years. It ships with the following library versions:
- gdal 3.6.2+dfsg-1+b
- proj 9.1.1-1+b1
- eccodes 2.28.0-1

Meanwhile, the current version thresholds in CLIMADA develop are:
- gdal>=3.6
- proj>=9.1
- eccodes>=2.27,<2.28

So this means that there could be full alignment if #678 were resolved which would be awesome!

My question is the following:
Do you think that it could be feasible to keep the alignment for these dependencies in place for as long as possible, ideally 1.5 years until the next Debian base image version is already appearing on the horizon and an upgrade of these dependencies can be planned and coordinated together?

It's completely clear that there could be very good reasons to upgrade these dependencies before that, I just wanted to drag the focus to the current situation which is very beneficial for the container team so that it could last for as long as reasonable.

Thanks in advance! Peter

**Comment by peanutfun** [comment]
Thank you for your input, @petermkr! As this is not urgent and a rather convoluted topic, I'll take the liberty to move this to the Discussions

**Comment by peanutfun** [comment]
With #745 we are slowly moving towards automated testing on GitHub. Since resources there are (currently) unlimited, we could always add additional workflows to check compatibility with Debian. I very much like your idea of keeping CLIMADA compatible to the latest stable release of Debian and/or Ubuntu.

What I think will come well into play here is the Docker image for CLIMADA. We could set up a separate pipeline which
* Builds a CLIMADA Docker image based on Debian Bookworm
* Runs unit tests
* ... (publishes the image on Docker Hub?)

Is the image of CLIMADA you want to publish based on Debian, too? Then integrating this into a workflow should not be a large issue. Note however, that it might make more sense to add the image build instructions, in particular the `Dockerfile`, to *this* repository, rather than using a separate one.

**Reply by petermkr** [reply]
Thanks for your reply!
It is indeed closely linked with the CLIMADA Docker image... this is exactly the image where having compatible dependencies would be great, and this is precisely the image which @ThomasRoosli and coworkers would contribute at some point.

I agree to the proposed workflow of image building, running the tests and publishing completely.

Regarding the location, the only thing to consider when using this repo here is how petals tests and libraries would come into play.

PS: Thanks for moving the issue to discussions, didn't know there was this feature, but this is definitely the right spot.
